### PR TITLE
Fix Build Failure

### LINKED ISSUE #312: Investigate why build failure is occurring only on mac
Investigate sudden failures after previously they were passing as seen here (there are two runs on the same commit):
https://github.com/IntelliTect/EssentialCSharp/runs/7528181670

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**
fixes #312

### CODE REVIEW

**Comment by BenjaminMichaelis** [comment]
Updating checkout to v3 and setup dotnet to v2

**Comment by BenjaminMichaelis** [comment]
After adding the props file in, it caused the ubuntu and windows builds to fail which is interesting since that means adding in a props file fails on dotnet 5 (the build is here: https://github.com/IntelliTect/EssentialCSharp/runs/7532253715?check_suite_focus=true) but after adding in a global.json and installing the dotnet 6 sdk onto the machine, then it passes for all builds. (https://github.com/IntelliTect/EssentialCSharp/runs/7532292828?check_suite_focus=true).

File: .github/workflows/Branch v9.0.yml [hunk 1]
```diff
@@ -17,13 +17,14 @@ jobs:
...
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup dotnet SDK
-        uses: actions/setup-dotnet@v1
+        uses: actions/setup-dotnet@v2
         with:
           dotnet-version: | 
             3.1.x
             5.0.x
+            6.0.x
```
Comment (Keboo) [inline .github/workflows/Branch v9.0.yml]:
```suggestion
            3.1.x
            5.x
            6.x
```

File: global.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (Keboo) [inline global.json:1]: Do we still need this after the Directory.Build.props change?

File: src/Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1 [hunk 3]
```diff
@@ -23,7 +23,8 @@ try {
     Get-Item "$PSScriptRoot/$ConsoleProgramProjectName" -ErrorAction Ignore | Remove-Item  -Recurse
     Set-PSDebug -Trace $traceLevel
     dotnet new Console --output "$PSScriptRoot/$ConsoleProgramProjectName"
-  
+    Set-Content ../../../../Chapter10/ProcessExitTestProgram/Directory.Build.props '<Project></Project>'
```
Comment (Keboo) [inline src/Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1]:
```suggestion
    Set-Content $PSScriptRoot/$ConsoleProgramProjectName/Directory.Build.props '<Project></Project>'
```

**Comment by BenjaminMichaelis** [comment]
So currently we do need a global.json because we ended up with failures on dotnet 5 of "warningCS8632:Theannotationfornullablereferencetypesshouldonlybeusedincodewithina'#nullable'annotationscontext.". I have a suspicion this is because with the directory props file being empty, when the csproj file is being generated, it by default has nullability disabled. I am testing this theory by adding it to be enabled in the pwsh script here: https://github.com/IntelliTect/EssentialCSharp/pull/318 . If it passes, I will get to drop the globl.json in this pr and just instead add nullability to the pwsh script for the props file.

File: src/Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1 [hunk 4]
```diff
@@ -23,7 +23,8 @@ try {
     Get-Item "$PSScriptRoot/$ConsoleProgramProjectName" -ErrorAction Ignore | Remove-Item  -Recurse
     Set-PSDebug -Trace $traceLevel
     dotnet new Console --output "$PSScriptRoot/$ConsoleProgramProjectName"
-  
+    Set-Content ../../../../Chapter10/ProcessExitTestProgram/Directory.Build.props '<Project><PropertyGroup><Nullable>enable</Nullable></PropertyGroup></Project>'
```
Comment (Keboo) [inline src/Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1]:
```suggestion
    Set-Content ../../../../Chapter10/ProcessExitTestProgram/Directory.Build.props '<Project><PropertyGroup><Nullable>enable</Nullable></PropertyGroup></Project>'
```
```suggestion
    Set-Content $PSScriptRoot/$ConsoleProgramProjectName/Directory.Build.props '<Project><PropertyGroup><Nullable>enable</Nullable></PropertyGroup></Project>'
```

**Review by Keboo — APPROVED** [review]

**Pull request merged**